sdk/python/lib/pulumi/dynamic/__main__.py

```diff
@@ -36,13 +39,23 @@
 _GRPC_CHANNEL_OPTIONS = [("grpc.max_receive_message_length", _MAX_RPC_MESSAGE_SIZE)]
 
 
-_PROVIDER_CACHE: typing.Dict[str, ResourceProvider] = {}
+_PROVIDER_CACHE: Dict[str, ResourceProvider] = {}
 _PROVIDER_LOCK = Lock()
 
 
-def get_provider(props) -> ResourceProvider:
+def get_provider(props: Dict[str, Any], config: Dict[str, Any]) -> ResourceProvider:
+    # Ensure Settings are configured in the thread that calls get_provider
+    configure(
+        Settings(
+            project=os.environ.get("PULUMI_PROJECT", "project"),
+            # `stack` and `organization` are the default values for Settings.
+            # Ideally we'd like to get the actual values here and set them.
+            stack="stack",
+            organization="organization",
```

Comment:
Any reason we aren't doing/able to do that? Sorry this is probably a dumb question.

Reply:
We don't have a good mechanism to pass this information through to plugins. We could possibly sneak it through on the defaultHost.

Maybe this is something that should come via the (real) configure call?